src/pg_cron.c

```diff
@@ -906,96 +1048,259 @@ ManageCronTask(CronTask *task, TimestampTz currentTime)
 				break;
 			}
 
-			task->runId = RunCount++;
 			task->pendingRunCount -= 1;
-			task->state = CRON_TASK_START;
+			if (UseBackgroundWorkers)
+				task->state = CRON_TASK_BGW_START;
+			else
+				task->state = CRON_TASK_START;
 
 			RunningTaskCount++;
+
+			/* Add new entry to audit table. */
+			SetCurrentStatementStartTimestamp();
+			StartTransactionCommand();
+			PushActiveSnapshot(GetTransactionSnapshot());
```

Comment:
These lines could maybe be moved into NextRunId since NextRunId is not used in any other context and we also put them in InsertOrUpdateJobRunDetail.